src/main/server/ServerClient.java

```
@@ -0,0 +1,55 @@
+package server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.Optional;
+
+public class ServerClient implements Client {
+
+  private final Socket clientConnection;
+  private final BufferedReader reader;
+  private final OutputStream writer;
+
+  public ServerClient(Socket clientConnection) throws IOException {
+    this.clientConnection = clientConnection;
+    this.reader = setupReader();
+    this.writer = setupWriter();
+  }
+
+  public Boolean isConnected() {
+    return !clientConnection.isClosed();
+  }
+
+  public Optional<String> readFrom() throws IOException {
+    String read = reader.readLine();
+    if (read != null) { return Optional.of(read); }
+    return Optional.empty();
+  }
+
+  public void sendTo(String data) {
+    try {
+      writer.write(data.concat("\n").getBytes());
+    } catch (IOException e) {
+      e.printStackTrace();
```

Do we want to swallow all exceptions and print to the console instead of dealing with the exception?